### ISSUE TITLE
custom-learning-office-365 manual installation error

### ISSUE THREAD

**Issue by fabizkk** [pnp/custom-learning-office-365#620]
Dear Team,

I am getting the below error while loading the page after completing the manual configuration.

TypeError: Cannot read properties of undefined (reading '0')
in t in t in n in t in t in t in t in t in section in div in div in t in Styledt in t in n in t in t in t

____________________________________________________________________
while running the configuration command ( -SiteAdminOnly) I am getting the below error.

Microsoft collects active usage data from your organization’s use of Microsoft 365 learning pathways and the use of Microsoft’s online content. Microsoft will use this data to help improve the future Microsoft 365 learning pathways solutions. To learn more about Microsoft privacy policies see https://go.microsoft.com/fwlink/?LinkId=521839. If you would like to opt out of this data collection, please type Ctrl-C to stop this script and see Readme file ("Disabling Telemetry Collection section") for instructions on how to opt out.

Press Enter to Continue:

Add-PnPPageSection : Cannot bind parameter 'Page'. Cannot convert the "OfficeDevPnP.Core.Pages.ClientSidePage" value of type
"OfficeDevPnP.Core.Pages.ClientSidePage" to type "PnP.PowerShell.Commands.Base.PipeBinds.PagePipeBind".
At C:\Users\fabis.kandy\Downloads\custom-learning-office-365-main\custom-learning-office-365-main\installation\M365lpConfiguration.ps1:129 char:54
+     $clvSection = Add-PnPClientSidePageSection -Page $clvPage -Sectio ...
+                                                      ~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Add-PnPPageSection], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,PnP.PowerShell.Commands.Pages.AddPageSection

.WARNING: The command 'Get-PnPAvailablePageComponents' is obsolete. Use Get-PnPPageComponent -Page -ListAvailable
Microsoft 365 learning pathways web parts found
Add-PnPPageWebPart : Cannot bind parameter 'Page'. Cannot convert the "OfficeDevPnP.Core.Pages.ClientSidePage" value of type
"OfficeDevPnP.Core.Pages.ClientSidePage" to type "PnP.PowerShell.Commands.Base.PipeBinds.PagePipeBind".
At C:\Users\fabis.kandy\Downloads\custom-learning-office-365-main\custom-learning-office-365-main\installation\M365lpConfiguration.ps1:151 char:36
+     Add-PnPClientSideWebPart -Page $clvPage -Component "Microsoft 365 ...
+                                    ~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Add-PnPPageWebPart], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,PnP.PowerShell.Commands.WebParts.AddWebPart

Set-PnPPage : Cannot bind parameter 'Identity'. Cannot convert the "OfficeDevPnP.Core.Pages.ClientSidePage" value of type
"OfficeDevPnP.Core.Pages.ClientSidePage" to type "PnP.PowerShell.Commands.Base.PipeBinds.PagePipeBind".
At C:\Users\fabis.kandy\Downloads\custom-learning-office-365-main\custom-learning-office-365-main\installation\M365lpConfiguration.ps1:152 char:37
+     Set-PnPClientSidePage -Identity $clvPage -Publish
+                                     ~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Set-PnPPage], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,PnP.PowerShell.Commands.Pages.SetPage

Add-PnPPageSection : Cannot bind parameter 'Page'. Cannot convert the "OfficeDevPnP.Core.Pages.ClientSidePage" value of type
"OfficeDevPnP.Core.Pages.ClientSidePage" to type "PnP.PowerShell.Commands.Base.PipeBinds.PagePipeBind".
At C:\Users\fabis.kandy\Downloads\custom-learning-office-365-main\custom-learning-office-365-main\installation\M365lpConfiguration.ps1:165 char:54
+     $claSection = Add-PnPClientSidePageSection -Page $claPage -Sectio ...
+                                                      ~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Add-PnPPageSection], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,PnP.PowerShell.Commands.Pages.AddPageSection

Add-PnPPageWebPart : Cannot bind parameter 'Page'. Cannot convert the "OfficeDevPnP.Core.Pages.ClientSidePage" value of type
"OfficeDevPnP.Core.Pages.ClientSidePage" to type "PnP.PowerShell.Commands.Base.PipeBinds.PagePipeBind".
At C:\Users\fabis.kandy\Downloads\custom-learning-office-365-main\custom-learning-office-365-main\installation\M365lpConfiguration.ps1:166 char:36
+     Add-PnPClientSideWebPart -Page $claPage -Component "Microsoft 365 ...
+                                    ~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Add-PnPPageWebPart], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,PnP.PowerShell.Commands.WebParts.AddWebPart

Set-PnPPage : Cannot bind parameter 'Identity'. Cannot convert the "OfficeDevPnP.Core.Pages.ClientSidePage" value of type
"OfficeDevPnP.Core.Pages.ClientSidePage" to type "PnP.PowerShell.Commands.Base.PipeBinds.PagePipeBind".
At C:\Users\fabis.kandy\Downloads\custom-learning-office-365-main\custom-learning-office-365-main\installation\M365lpConfiguration.ps1:167 char:37
+     Set-PnPClientSidePage -Identity $claPage -Publish
+                                     ~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Set-PnPPage], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,PnP.PowerShell.Commands.Pages.SetPage

**Comment by amitkh22** [comment]
Hi @fabizkk, thank you for reporting this and providing the details. Please allow us sometime to check this. We will get back to you at the earliest.

**Comment by andreaskelly** [comment]
Hey,
I have exactly the same issues.

@dcashpeterson 
1. yes
2. yes
3. yes
4. Pnp.PowerShell 1.9.0 , PowerShell 5.1.17763.2268

Cheers

**Comment by fabizkk** [comment]
> 2\. Are you authenticating as an account with SP Admin rights?

The issue has been resolved, as mentioned admin account was the issue.

**Comment by andreaskelly** [comment]
![learningpathwaysscripterror](https://user-images.githubusercontent.com/97537235/149080282-aae9b42a-47b7-49ed-ab84-cded33ea6c99.PNG)

I still get errors, unfortunately.

**Comment by dcashpeterson** [comment]
@fabizkk I am glad that worked for you and your issue is resolved.

@andreaskelly It looks like you have found something. :) The installation script in the repo was built using the SharePointPnPPowerShellOnline module not PnP.PowerShell. As PnP.PowerShell has evolved some of the commands have changed and because of that our install script isn't working with the latest version of PnP.PowerShell. I am marking this as a bug and we will work on it. In the mean time if you have the ability to install and use the previous version SharePointPnPPowerShellOnline module that should work to get you up and running. Thanks for uncovering this.

**Comment by andreaskelly** [comment]
Thank you @dcashpeterson ! I got it running, great.

**Comment by dcashpeterson** [comment]
@andreaskelly Just to circle back on this. We reviewed the installation script and made some small changes. The script needs to be run using the SharePointPnPPowerShellOnline Module and NOT PnP.PowerShell module. We made the choice not to update it as PnP.PowerShell requires an app registration to run with a lot of privileges and most of our users are not admins with the level of privileges to install that. Thanks for bringing this to our attention.
